nncf/quantization/algorithms/weight_compression/weight_lowering.py

```diff
@@ -121,6 +127,12 @@ def do_integer_quantization(
 
     if group_size != -1:
         # weights are reshaped from [a1, r, a2] to [a1, r//gs, gs, a2]
+        if isinstance(reduction_axis, tuple) and len(reduction_axis) == 1:
+            reduction_axis = reduction_axis[0]
+        if not isinstance(reduction_axis, int):
+            raise NotImplementedError(
```

Comment:
what about moving this to `reshape_weight_for_grouped_quantization` ?
you won't need to do it every time before calling this method